champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/axe/Leap.java

```diff
@@ -79,6 +89,25 @@ public void doLeap(Player player, boolean wallkick) {
 
         player.getWorld().spawnEntity(player.getLocation(), EntityType.LLAMA_SPIT);
         player.getWorld().playSound(player.getLocation(), Sound.ENTITY_BAT_TAKEOFF, 2.0F, 1.2F);
+        canTakeFall.put(player.getUniqueId(), true);
+    }
+
+    @EventHandler
+    public void reduceFallDamage(CustomDamageEvent event) {
+        if (event.getCause() != EntityDamageEvent.DamageCause.FALL) return;
+
+
+        Player player = (Player) event.getDamagee();
+        UUID playerId = player.getUniqueId();
+
+        if (canTakeFall.containsKey(playerId) && canTakeFall.get(playerId)) {
```

Comment:
this is fundamentally flawed. What if they dont leap high enough to take fall damage? Theyve essentially stored a -6 damage charge the next time they do happen to take damage

Reply:
Same applies to the other skill changes.